### PR TITLE
Catch the client connection "Close normal" error

### LINKED ISSUE #49: Error when client disconnects
See [this Sentry issue](https://sentry.io/share/issue/662aa9f33d2548e1b4001e3793480fa3/)

### PR DESCRIPTION
**Pull request by mpass99**
Closes  #49

The loop that reads the stdin notes that the client has left the connection, just before our close handler does.
Now we catch this case.

### CODE REVIEW

File: internal/api/websocket.go [hunk 1]
```diff
@@ -76,7 +76,11 @@ func (cr *codeOceanToRawReader) readInputLoop(ctx context.Context) {
 		case <-readMessage:
 		}
 
-		if err != nil {
+		if err != nil && websocket.IsCloseError(err, websocket.CloseNormalClosure) {
+			log.Debug("ReadInputLoop: The client closed the connection!")
+			// The close handler will do something soon.
```
Comment (MrSerth) [inline internal/api/websocket.go:81]: We are experiencing a race condition here, right?

**Review by MrSerth — APPROVED** [review]
Pretty simple fix, thanks!

File: internal/api/websocket.go (hunk 1 above)
Comment (mpass99) [reply internal/api/websocket.go:81]: Yes

Comment (MrSerth) [reply internal/api/websocket.go:81]: Alright. I think this handling is fine.

**Comment by mpass99** [comment]
> Pretty simple fix, thanks!

Apart from the fact that this additional condition was exactly the mü for a too high cognitive complexity (for our linter), yeah...

**Review by MrSerth — APPROVED** [review]
> > Pretty simple fix, thanks!
> 
> Apart from the fact that this additional condition was exactly the mü for a too high cognitive complexity (for our linter), yeah...

Yes, I noticed that as well. Your solution is smart even when it took me a moment to get it (which is not related to the new version but rather the diff and the changes I compared).

**Pull request merged**